doc/explanation/configure-storage.rst

```diff
@@ -0,0 +1,177 @@
+updated.
+
+amd64 systems use GRUB as the bootloader. amd64 systems can boot in either UEFI
+or legacy (sometimes called "BIOS") mode (many systems can be configured to
+boot in either mode) and the bootloader is located completely differently in
+the two modes.
+
+Legacy mode
+-----------
+
+In legacy mode, the bootloader is read from the first "sector" of a hard drive
+(exactly which hard drive is up to the system firmware, which can usually be
+configured in a vendor-specific way). The installer will write GRUB to the
+start of all disks selected as a boot devices. As GRUB does not entirely fit
+in one sector, a small unformatted partition is needed at the start of the
+disk, which will automatically be created when a disk is selected as a boot
+device (a disk with an existing GPT partition table can only be used as a boot
+device if it has this partition).
+
+UEFI mode
+---------
+
+In UEFI mode, the bootloader loaded from a "EFI System Partition" (ESP), which
+is a partition with a particular type GUID. The installer automatically creates
+a 512MiB ESP on a disk when it is selected as a boot device and will install
+GRUB there (a disk with an existing partition table can only be used as a boot
+device if it has an ESP -- bootloaders for multiple operating systems can be
+installed into a single ESP). UEFI defines a standard way to configure the way
+in which the operating system is chosen on boot, and the installer uses this to
+configure the system to boot the just-installed operating system. One of the
+ESPs must be mounted at ``/boot/efi``.
+
+Supported arm64 servers boot using UEFI, and are configured the same way as an
+UEFI-booting amd64 system.
+
+ppc64el systems also load their bootloader (Petitboot, a small linux kernel)
+from a "PReP" partition with a special flag, so in most ways they are similar
+to a UEFI system. The installer only supports one PReP partition at this time.
+
+Limitations and workarounds
+===========================
+
+Currently, the installer cannot **edit** partition tables. You can use existing
+partitions or reformat a drive entirely, but you cannot (for example) remove a
+large partition and replace it with two smaller ones.
+
+The installer allows the creation of LVM volume groups and logical volumes and
+MD raid devices, but does not allow tweaking of the parameters -- for example,
+all logical volumes are linear and all MD raid devices use the default metadata
+format (1.2).
+
+These limits can both be worked around in the same way: drop to a shell and use
+the usual shell commands to edit the partition table or create the LV or RAID
+with desired parameters, and then select these partitions or devices as mount
+points in the installer. Any changes you make while the installer is running
+but before altering the storage configuration will reflected in the installer.
+
+The installer cannot yet configure iSCSI mounts, ZFS at all, or btrfs
```

Comment:
Let's remove the mention of ZFS, we'll be able to offer a limited form in 23.10 (but not through the TUI)

Reply:
So can it configure ZFS now? Or not yet?

Reply:
yes, but the 23.10 answer will be:
desktop -> a checkbox in advanced
desktop and server -> an option in autoinstall (https://github.com/canonical/subiquity/pull/1763/files#diff-d2236521d63f00962c3248afbee5d33b032b8dfd54c29e5310fe492655fa8267R409)

Reply:
Understood, thanks for the clarification